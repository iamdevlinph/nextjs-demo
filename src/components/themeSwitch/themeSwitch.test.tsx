import { mount, shallow } from 'enzyme';
import React from 'react';

import ThemeSwitch from './themeSwitch';

describe('ThemeSwitch Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ThemeSwitch />);
    expect(wrapper).toMatchSnapshot();
  });

  // TODO test trigger
  // it('should trigger toggleDarkMode onChange', () => {
  //   const wrapper = mount(<ThemeSwitch />);
  //   const mockFn = jest.fn();

  //   wrapper.find('input').simulate('change');
  // });
});
