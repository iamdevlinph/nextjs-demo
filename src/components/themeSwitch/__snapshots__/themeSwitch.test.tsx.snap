// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitch Component should render correctly 1`] = `
<ReactSwitch
  activeBoxShadow="0 0 2px 3px #3bf"
  boxShadow={null}
  checked={false}
  checkedIcon={
    <i
      className="icon-moon"
    />
  }
  className="text-black border border-black dark:text-white dark:border-gray-500"
  disabled={false}
  height={28}
  offColor="#f7fafc"
  offHandleColor="#1a202c"
  onChange={[Function]}
  onColor="#1a202c"
  onHandleColor="#f7fafc"
  uncheckedIcon={
    <i
      className="icon-sun"
    />
  }
  width={56}
/>
`;
